src/components/token/TokenBalances.js

```diff
@@ -0,0 +1,327 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import TokenBalancesTable from './TokenBalancesTable';
+import tokensApi from '../../api/tokensApi';
+import { get, last, find, isEmpty } from 'lodash';
+import PaginationURL from '../../utils/pagination';
+import { withRouter } from "react-router-dom";
+import ErrorMessageWithIcon from '../error/ErrorMessageWithIcon'
+import TokenAutoCompleteField from './TokenAutoCompleteField';
+import helpers from '../../utils/helpers';
+
+/**
+ * Displays custom tokens in a table with pagination buttons and a search bar.
+ */
+class TokenBalances extends React.Component {
+  /**
+   * Structure that contains the attributes that will be part of the page URL
+   */
+  pagination = new PaginationURL({
+    'sortBy': { required: false },
+    'order': { required: false },
+  });
+
+  constructor(props) {
+    super(props);
+
+    /**
+     * tokenBalances: List of token balances currently being rendered.
+     *         Each token balance element must have the fields: address, locked_balance, unlocked_balance, total, token_id and sort.
+     *         id, name, symbol are strings; nft is boolean; transaction_timestamp is long.
+     *         Sort is an array with two strings, The value is given by ElasticSearch and it is passed back when we want to change page
+     * hasAfter: Indicates if a next page exists
+     * hasBefore: Indicates if a previous page exists
+     * sortBy: Which field to sort (uid, name, symbol)
+     * order: If sorted field must be ordered asc or desc
+     * page: Current page. Used to know if there is a previous page
+     * pageSearchAfter: Calculates the searchAfter param that needs to be passed to explorer-service in order to get the next/previous page.
+     *                  We use this array to store already-calculated values,
+     *                  so we do not need to recalculate them if user is requesting an already-navigated page.
+     * loading: Initial loading, when user clicks on the Tokens navigation item
+     * isSearchLoading: Indicates if search results are being retrieved from explorer-service
+     * calculatingPage: Indicates if next page is being retrieved from explorer-service
+     * error: Indicates if an unexpected error happened when calling the explorer-service
+     * tokenBalanceInformationError: Indicates if an unexpected error happened when calling the token balance information service 
+     * maintenanceMode: Indicates if explorer-service or its downstream services are experiencing problems. If so, maintenance mode is enabled as
```

Comment:
Maybe this description is slightly misleading.

By reading this, I could think that there is really a circuit breaking mechanism implemented that will enable the `maintenanceMode` automatically if it detects problems.

But in reality we just have the flag in Unleash which is activated manually, right?

Reply:
You are right, refactored the comment on 724795423ba5e17336b98377b740b163543287bd